### ISSUE TITLE
Russian filename modified is modified when installing addon

### ISSUE THREAD

**Issue by feerrenrut** [nvaccess/nvda#10690]
### Steps to reproduce:
1. Unzip the attached addon, and inspect its files. Note the names of the files.
1. Install the attached addon (first renaming it to remove the .zip)
1. Inspect your `userConfig/addons/Check_Russian_language` folder, note the names of the files.

### Actual behavior:
File called `Римские цифры.txt` becomes `É¿¼ß¬¿Ñ µ¿Σαδ.txt`
There is no related error in the log.

### Expected behavior:
File called `Римские цифры.txt` should not be renamed.

#### NVDA version:
2019.3 beta 1

#### Name and version of other software in use when reproducing the issue:
Use the following addon (but first rename to `Check_Russian_language-V1.0.nvda-addon`, removing the `.zip` part)
[Check_Russian_language-V1.0.nvda-addon.zip](https://github.com/nvaccess/nvda/files/4071735/Check_Russian_language-V1.0.nvda-addon.zip)

**Comment by zstanecic** [comment]
Hi Reef,

We have the problem called the encoding.

It is the classic encoding windows problem, and file name should be re-encoded to utf-8.

**Comment by feerrenrut** [comment]
I haven't run into this before, why is the filename correct if unzipped with 7-zip but not if unzipped by NVDA?

**Comment by feerrenrut** [comment]
This addon was sent to us to demonstrate the issue.
After unzipping the addon with 7-zip, I re-zipped the original files with 7-zip (using zip method) and renamed to `*.nvda-addon`. This now installs in NVDA without changing the filename.

I'm going to close this issue. If anyone has further insight into how the original developer could avoid this problem in the future, please leave a comment.

**Comment by tspivey** [comment]
7-Zip sets bit 11 of the zip header flags, which sets UTF-8 for the filename field. Python's zipfile module knows how to handle that.
The original zip file doesn't, instead using extra field 0x7075 to store a UTF-8 version of the filename.

I don't know which archiver was used to create this file, but the solution is to use something else.

**Comment by RussianSiberian** [comment]
Good luck!

I'm so ashamed that I stole your time for my problem, although I could solve it by yourself. Excuse me. 
When I started to write the add-on’s for NVDA, I have studied a lot of documentation. One of them was at:
https://www.nvaccess.org/files/nvda/documentation/developerGuide.html
There says how to use non-ASCII File Names in Zip Archives.
I didn't notice the difference Using and not using "cu=on" method parameter in 7-zip. Russian text is displayed correctly in both cases NVDA on Python 2. And I forgot about it. 
But when NVDA switched to Python 3, I was very surprised. I didn't know what the error may be on my part.
Once again, I apologize.

Kind regards Alexandr.
